examples/features/mesh_light.py

```diff
@@ -67,9 +66,12 @@ def update_params():
             (int(c[0] * 255), int(c[1] * 255), int(c[2] * 255), int(c[3] * 255)), i
         )
         visual.set_light_pos(light_pos[i], i)
-        visual.set_material_params(material[i], i)  # for ambient, specular, diffuse, and emission.
-    visual.set_shine(shine.value)
-    visual.set_emit(emit.value)
+    visual.set_ambient_params(material['ambient_params'])
+    visual.set_diffuse_params(material['diffuse_params'])
+    visual.set_specular_params(material['specular_params'])
+    visual.set_emission_params(material['emission_params'])
+    visual.set_shine(material['shine'].value)
```

Comment:
We can probably tweak the Visual or Prop class to make it work without the explicit `.value` here, for consistency with the `vec3` parameters.

Reply:
Methods like `visual.set_shine()` currently call `visual.shine = a_float` under the hood (in `__setattr__()`). This is the magic I had mentioned earlier and that we could revisit. In any case, I think we could tweak this generic logic so that scalar setter methods like `set_shine()` work with either a float or an Out. So you could do either `visual.set_shine(Out(value))` or `visual.set_shine(value)`.

I agree that the renaming can be done in a separate PR.